src/D2L.CodeStyle.Analyzers/Diagnostics.cs

```diff
@@ -609,5 +609,59 @@ public static class Diagnostics {
 			isEnabledByDefault: true,
 			description: "RPC parameters must be of a valid type."
 		);
+
+		public static readonly DiagnosticDescriptor ReflectionSerializer_ConstructorParameterCannotBeDeserialized = new DiagnosticDescriptor(
+			id: "D2L0081",
+			title: "[ReflectionSerializer] attribute requires that constructor parameters have a corresponding property of the same name.",
+			messageFormat: "The '{0}' parameter is missing a corresponding property of the same name. [ReflectionSerializer] requires that constructor parameters have a corresponding property of the same name in order to deserialize.",
+			category: "Correctness",
+			defaultSeverity: DiagnosticSeverity.Error,
+			isEnabledByDefault: true
+		);
+
+		public static readonly DiagnosticDescriptor ReflectionSerializer_Class_NoPublicConstructor = new DiagnosticDescriptor(
+			id: "D2L0082",
+			title: "[ReflectionSerializer] attribute requires classes to have a public constructor.",
+			messageFormat: "The type has no public constructors. [ReflectionSerializer] attribute requires classes to have a public constructor in order to deserialize.",
+			category: "Correctness",
+			defaultSeverity: DiagnosticSeverity.Error,
+			isEnabledByDefault: true
+		);
+
+		public static readonly DiagnosticDescriptor ReflectionSerializer_Class_MultiplePublicConstructors = new DiagnosticDescriptor(
+			id: "D2L0083",
+			title: "[ReflectionSerializer] attribute requires classes to have a single public constructor.",
+			messageFormat: "The type has too many public constructors. [ReflectionSerializer] attribute requires classes to have a single public constructor in order to deterministically deserialize",
+			category: "Correctness",
+			defaultSeverity: DiagnosticSeverity.Error,
+			isEnabledByDefault: true
+		);
+
+		public static readonly DiagnosticDescriptor ReflectionSerializer_Record_NoPublicConstructor = new DiagnosticDescriptor(
+			id: "D2L0084",
+			title: "[ReflectionSerializer] attribute requires records to have an inline primary constructor or a public constructor.",
+			messageFormat: "The type has no public constructors. [ReflectionSerializer] attribute requires records to have an inline primary constructor or a public constructor.",
+			category: "Correctness",
+			defaultSeverity: DiagnosticSeverity.Error,
+			isEnabledByDefault: true
+		);
+
+		public static readonly DiagnosticDescriptor ReflectionSerializer_Record_MultiplePublicConstructors = new DiagnosticDescriptor(
+			id: "D2L0085",
+			title: "[ReflectionSerializer] attribute requires records to have just an inline primary constructor or a single public constructor.",
+			messageFormat: "The type has too many public constructors. [ReflectionSerializer] attribute requires records to have just an inline primary constructor or a single public constructor.",
+			category: "Correctness",
+			defaultSeverity: DiagnosticSeverity.Error,
+			isEnabledByDefault: true
+		);

```

Comment:
Consider combining these? Maybe just wording like "For records this includes the inline constructor." would suffice to make the class ones work for either?

Reply:
Done